x-pack/plugins/enterprise_search/public/applications/app_search/components/meta_engine_creation/meta_engine_creation_logic.ts

```diff
@@ -0,0 +1,131 @@
+  ): { selectedIndexedEngineNames: MetaEngineCreationValues['selectedIndexedEngineNames'] };
+  submitEngine(): void;
+}
+
+export const MetaEngineCreationLogic = kea<
+  MakeLogicType<MetaEngineCreationValues, MetaEngineCreationActions>
+>({
+  path: ['enterprise_search', 'app_search', 'meta_engine_creation_logic'],
+  actions: {
+    fetchIndexedEngineNames: (page = DEFAULT_META.page.current) => ({ page }),
+    onEngineCreationSuccess: true,
+    setIndexedEngineNames: (indexedEngineNames) => ({ indexedEngineNames }),
+    setRawName: (rawName) => ({ rawName }),
+    setSelectedIndexedEngineNames: (selectedIndexedEngineNames) => ({ selectedIndexedEngineNames }),
+    submitEngine: () => null,
+  },
+  reducers: {
+    indexedEngineNames: [
+      [],
+      {
+        setIndexedEngineNames: (_, { indexedEngineNames }) => indexedEngineNames,
+      },
+    ],
+    rawName: [
+      '',
+      {
+        setRawName: (_, { rawName }) => rawName,
+      },
+    ],
+    selectedIndexedEngineNames: [
+      [],
+      {
+        setSelectedIndexedEngineNames: (_, { selectedIndexedEngineNames }) =>
+          selectedIndexedEngineNames,
+      },
+    ],
+  },
+  selectors: ({ selectors }) => ({
+    name: [() => [selectors.rawName], (rawName: string) => formatApiName(rawName)],
+  }),
+  listeners: ({ values, actions }) => ({
+    fetchIndexedEngineNames: async ({ page }) => {
+      const { http } = HttpLogic.values;
+      let response;
+
+      try {
+        response = (await http.get('/api/app_search/engines', {
+          query: { type: 'indexed', 'page[current]': page, 'page[size]': DEFAULT_META.page.size },
+        })) as { results: EngineDetails[]; meta: Meta };
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+
+      if (response) {
+        const engineNames = response.results.map((result) => result.name);
+        actions.setIndexedEngineNames([...values.indexedEngineNames, ...engineNames]);
+
+        if (page < response.meta.page.total_pages) {
+          actions.fetchIndexedEngineNames(page + 1);
+        }
+      }
+    },
+    onEngineCreationSuccess: () => {
+      const { name } = values;
+      const { navigateToUrl } = KibanaLogic.values;
+      const enginePath = generatePath(ENGINE_PATH, { engineName: name });
+
+      setQueuedSuccessMessage(META_ENGINE_CREATION_SUCCESS_MESSAGE);
+      navigateToUrl(enginePath);
+    },
+    submitEngine: async () => {
+      const { http } = HttpLogic.values;
+      const { name, selectedIndexedEngineNames } = values;
+
+      const body = JSON.stringify({
+        name,
+        type: 'meta',
+        source_engines: selectedIndexedEngineNames,
+      });
+
+      let response;
+      try {
+        response = await http.post('/api/app_search/engines', { body });
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+
+      if (response) {
+        actions.onEngineCreationSuccess();
+      }
```

Comment:
I'm a little confused here as to why `let response` need to be defined outside of the `try` / why we need to check for a response?

```suggestion
      try {
        const response = await http.post('/api/app_search/engines', { body });
        actions.onEngineCreationSuccess();
      } catch (e) {
        flashAPIErrors(e);
      }
```

Reply:
I don't like wrapping stuff in the try that I don't expect to throw errors.  `flashAPIErrors` will only handle errors coming from our API, if we're worried about `actions.onEngineCreationSuccess` throwing an exception as well we should wrap that in a seperate try statement.

Reply:
I can see the rationale behind that, but this is definitely a deviation from how we've been writing try/catches and API calls across both AS and WS. My concern here is one of consistency; a developer coming across this file will be confused (as I was) as to why this was written differently from all our other logic files.

I'd definitely suggest putting this before the frontend channel and asking for thoughts as a team. My 2c personally is that the redefinition here is not worth it. Even if we don't expect anything after the API call to throw an error, `flashAPIErrors` has been written robustly enough to handle non-API/JS errors - it re-throws them instead of displaying them in a flash message (so essentially non-API errors behave the same as if they were not within a try/catch). IMO, easier dev readability and succinctness trumps precision in this case.

Reply:
> flashAPIErrors has been written robustly enough to handle non-API/JS errors
This is great!

> IMO, easier dev readability and succinctness trumps precision in this case.
I agree. And consistency is also a strong argument for me.

Reply:
I agree with Byron, but since Constance states "flashAPIErrors has been written robustly enough to handle non-API/JS errors - it re-throws them", then it sounds like it's safe to implement the more succinct syntax she suggests, so I vote for that.

Reply:
If the majority of the team would rather write our API calls with a try only around the API call as a best practice, I'm also totally fine with that - I would just want us to agree on that as a team + make that change consistently across the codebase if that makes sense.